File: 1260-shift-2d-grid/1260-shift-2d-grid.ts
function shiftGrid(grid: number[][], k: number): number[][] {
    const m = grid.length;
    const n = grid[0].length;
    const total = m * n;

    k = k % total;

    const ans: number[][] = Array.from({ length: m }, () => Array(n).fill(0));

    for (let i = 0; i < m; i++) {

        for (let j = 0; j < n; j++) {

            const oldIndex = i * n + j;
            const newIndex = (oldIndex + k) % total;
            const newRow = Math.floor(newIndex / n);
            const newCol = newIndex % n;
            
            ans[newRow][newCol] = grid[i][j];
        }
    }
    return ans;
}

File: 1260-shift-2d-grid/1260-shift-2d-grid.rs
impl Solution {
    pub fn shift_grid(grid: Vec<Vec<i32>>, k: i32) -> Vec<Vec<i32>> {
        let m = grid.len();
        let n = grid[0].len();
        let total = m * n;
        let k = (k as usize) % total;

        let mut ans = vec![vec![0; n]; m];

        for i in 0..m {
            for j in 0..n {
                let old_index = i * n + j;
                let new_index = (old_index + k) % total;
                let new_row = new_index / n;
                let new_col = new_index % n;
                
                ans[new_row][new_col] = grid[i][j];
            }
        }
        ans
    }
}